opc/resource_lbaas_load_balancer.go

```diff
@@ -0,0 +1,327 @@
+	}
+
+	result, err := lbClient.GetLoadBalancer(lb)
+	if err != nil {
+		// LoadBalancer does not exist
+		if client.WasNotFoundError(err) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading Load Balancer %s: %s", d.Id(), err)
+	}
+
+	if result == nil {
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("canonical_host_name", result.CanonicalHostName)
+	d.Set("cloudgate_capable", result.CloudgateCapable)
+	d.Set("enabled", getEnabledState(result.Disabled))
+	d.Set("description", result.Description)
+	d.Set("ip_network", result.IPNetworkName)
+	d.Set("name", result.Name)
+	d.Set("parent_load_balancer", result.ParentLoadBalancer)
+	d.Set("region", result.Region)
+	d.Set("scheme", result.Scheme)
+	d.Set("server_pool", result.OriginServerPool)
+	d.Set("uri", result.URI)
+
+	if err := setStringList(d, "balancer_vips", result.BalancerVIPs); err != nil {
+		return err
+	}
+	if err := setStringList(d, "permitted_clients", result.PermittedClients); err != nil {
+		return err
+	}
+	if err := setStringList(d, "permitted_methods", result.PermittedMethods); err != nil {
+		return err
+	}
+	if err := setStringList(d, "policies", result.Policies); err != nil {
+		return err
+	}
+	if err := setStringList(d, "tags", result.Tags); err != nil {
+		return err
+	}
+	return nil
+}
+
+func resourceOPCLoadBalancerUpdate(d *schema.ResourceData, meta interface{}) error {
+	lbaasClient, err := meta.(*Client).getLBaaSClient()
+	if err != nil {
+		return err
+	}
+	lbClient := lbaasClient.LoadBalancerClient()
+	lb := getLoadBalancerContextFromID(d.Id())
+
+	input := lbaas.UpdateLoadBalancerInput{
+		Name: d.Get("name").(string),
+	}
+
+	if enabled, ok := d.GetOk("enabled"); ok {
+		input.Disabled = getDisabledStateKeyword(enabled.(bool))
+	}
+
+	input.Description = updateOrRemoveStringAttribute(d, "description")
+	input.IPNetworkName = updateOrRemoveStringAttribute(d, "ip_network")
+	input.OriginServerPool = updateOrRemoveStringAttribute(d, "server_pool")
+	input.ParentLoadBalancer = updateOrRemoveStringAttribute(d, "parent_load_balancer")
+
+	input.PermittedClients = updateOrRemoveStringListAttribute(d, "permitted_clients")
+	input.PermittedMethods = updateOrRemoveStringListAttribute(d, "permitted_methods")
+	input.Policies = updateOrRemoveStringListAttribute(d, "policies")
+	input.Tags = updateOrRemoveStringListAttribute(d, "tags")
+
+	result, err := lbClient.UpdateLoadBalancer(lb, &input)
+	if err != nil {
+		return fmt.Errorf("Error updating LoadBalancer: %s", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", lb.Region, result.Name))
+
+	return resourceOPCLoadBalancerRead(d, meta)
+}
+
+func resourceOPCLoadBalancerDelete(d *schema.ResourceData, meta interface{}) error {
+	lbaasClient, err := meta.(*Client).getLBaaSClient()
+	if err != nil {
+		return err
+	}
+	lbClient := lbaasClient.LoadBalancerClient()
+	lb := getLoadBalancerContextFromID(d.Id())
+
+	if _, err := lbClient.DeleteLoadBalancer(lb); err != nil {
+		return fmt.Errorf("Error deleting LoadBalancer")
```

Comment:
Let's also write out the error that get's returned when we fail to delete the Load Balancer

Reply:
done